### ISSUE TITLE
Differences between the outputs of 2D and 3D simulations

### ISSUE THREAD

**Issue by talebimohammad** [ydluo/qdyn#4]
Dear developers of QDYN software,

I am a Ph.D. candidate of Earthquake-Seismology at the International Institute of Earthquake Engineering and Seismology (IIEES), Tehran, Iran. I am new in the field of earthquake simulation, so my questions are probably quite naive.

I began learning QDYN by running the "JP_2D_s.m" and “ JP_3d.m” files (without deep asperities) located in the ~/example/Tohoku folder.
To have the same parameters in simulations, I have decreased the solver accuracy of the 3D model to “1e-14”, same as that of 2D model.
My questions are motivated by checking some of the outputs of 2D and 3D simulations.
It seems to me that the results for 2D simulation are different from those of 3D ones. For example, I have compared the recurrence time-periods of the M9.1 event  and maximum slip rates (ot.v) for the models.
On the other hand, the results of the 3D-model for different sets of along-strike rupture lengths (p.L) are not stable. 

I would be very grateful if you could kindly explain how to fix these problems.

Looking forward to hearing from you.

Best regards,

Mohammad Talebi

**Comment by ydluo** [comment]
Hi Mohammad,

Thank you so much for your interest in QDYN.

It’s natural that 2D/3D results are different, it will be helpful if you can be more specific of the model your are using, i.e. scripts/parametrical settings.
As of the ‘unstable’, can you describe it in a little bit detail, what kind of instability do you have? that could duo to resection of the simulation (checking Lb/dx, etc),
or the fault itself can display a various of macroscopical behaviors


Yingdi 



------------

Yingdi LUO

Seismological Laboratory
MS 252-21
California Institute of Technology
Pasadena, CA 91125

Phone: +1-626-395-6971
Fax: +1-626-564-0715
Email: luoyd@gps.caltech.edu
	     luoyd@caltech.edu

**Comment by talebimohammad** [comment]
Hi Yingdi,

Thanks for your quick reply.

In fact, I know the general aspects of the natural differences between 2D and 3D simulations. My questions arise from the aim of performing 2D simulations in my case that is reducing the computational cost. Indeed, my comment was to see if there is a way to set the parameters of the friction law in 2D model so that observations (and/or maybe assumptions, like slip value or recurrence time) will be satisfied by 3D modelling or not.
But, according to your response, in order to satisfy the observations (and/or maybe assumptions) in 3D modelling, it needs to define new parameters of the 3D-model separately. Am I right?

On the other hand, for great earthquakes (M>8) like Tohoku, the degree of differences (not absolute values of differences) is probably negligible, but not for a major earthquake. For instance, as I have checked the maximum slip rate of an earthquake with magnitude ~6.1-6.3 (duration ~5-10 sec and maximum slip during co-seimic phase ~0.5-1 m), the degree of differences between the results of 2D and 3D simulations was ~1000. I mean, for this case, instead of an earthquake, I have got a creep model in 3D modelling by putting initial conditions from 2D simulation.

In the case of instability, it seems to me that it is mostly due to dividing the fault into sections. I set the parameters so that I have minimum value of Lb/dx greater than 1. Considering reducing the computational cost, my question is: if Lb/dx exceeds 1, how can we know whether the results is significant?

Also, I have enclosed the related scripts to my e-mail.

Thanks for your time and looking forward to hearing from you.

Kind regards,

Mohammad
